sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/crypto/_client.py

```diff
@@ -311,16 +314,16 @@ def decrypt(self, algorithm, ciphertext, **kwargs):
             )
 
         operation_result = self._client.decrypt(
-            vault_base_url=self._key_id.vault_url,
-            key_name=self._key_id.name,
-            key_version=self._key_id.version,
+            vault_base_url=self._key_id.vault_url if self._key_id else None,
+            key_name=self._key_id.name if self._key_id else None,
+            key_version=self._key_id.version if self._key_id else None,
             parameters=self._models.KeyOperationsParameters(
                 algorithm=algorithm, value=ciphertext, iv=iv, tag=tag, aad=aad
             ),
             **kwargs
         )
 
-        return DecryptResult(key_id=self.key_id, algorithm=algorithm, plaintext=operation_result.result)
+        return DecryptResult(algorithm=algorithm, plaintext=operation_result.result, key_id=self.key_id)
```

Comment:
These parameters can be returned to their original order in methods returning `*Result`s here and in the async client

Reply:
Also, have reverted the order in other *Results in this file and async one.